app/models/permission_set.rb

```diff
@@ -0,0 +1,34 @@
+class PermissionSet
+  include Enumerable
+
+  attr_reader :permissions
+  delegate :each, :last, to: :permissions
+
+  def self.for(username)
+    permissions = RolesDb.client.accounts_for(username).map do |account|
+      Permission.from(account)
+    end
+
+    new(permissions)
+  end
+
+  def initialize(permissions)
+    @permissions = Array(permissions)
+  end
+
+  def department_slugs
+    inject([]) do |slugs, permission|
```

Comment:
I *think* that you could use `each_with_object` instead of `inject` and you wouldn't need to return `slugs`.